tests/python/pants_test/engine/legacy/test_options_parsing.py

```diff
@@ -0,0 +1,51 @@
+# coding=utf-8
+# Copyright 2018 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import (absolute_import, division, generators, nested_scopes, print_function,
+                        unicode_literals, with_statement)
+
+import os
+import unittest
+from contextlib import contextmanager
+
+from pants.base.build_environment import get_buildroot
+from pants.engine.legacy.options_parsing import Options, OptionsParseRequest
+from pants.init.engine_initializer import EngineInitializer
+from pants_test.engine.util import init_native
+
+
+class TestEngineOptionsParsing(unittest.TestCase):
+
+  _native = init_native()
+
+  @contextmanager
+  def scheduler(self, build_root=None, native=_native):
+    build_root = build_root or get_buildroot()
+    work_dir = os.path.join(build_root, '.pants.d')
+    scheduler = EngineInitializer.setup_legacy_graph(
+      [],
+      work_dir,
+      build_file_imports_behavior='allow',
+      build_root=build_root,
+      native=native
+    )
+    yield scheduler.scheduler.new_session()
+
+  def test_options_parsing_request(self):
+    with self.scheduler() as scheduler:
```

Comment:
It might also be possible to test this rule in isolation with: https://github.com/pantsbuild/pants/blob/2e65f46e044ed6c9fce07a240615e7a96d8b7c27/tests/python/pants_test/engine/util.py#L25-L45

Reply:
ah, nice.

Reply:
worked out well!